src/stores/connectors/stores/create_key.go

```diff
@@ -2,177 +2,83 @@ package stores
-	logger := c.logger.With("vault_type", vaultType, "store_name", storeName)
+func (c *Connector) CreateKey(ctx context.Context, name, vault, secretStore string, allowedTenants []string, userInfo *auth.UserInfo) error {
+	logger := c.logger.With("name", name, "vault", vault, "secret_store", secretStore)
 	logger.Debug("creating key store")
 
-	store, err := c.createKeyStore(storeName, vaultType, specs)
-	if err != nil {
-		return err
-	}
-
-	c.mux.Lock()
-	c.stores[storeName] = &entities.StoreInfo{
-		AllowedTenants: allowedTenants,
-		Store:          store,
-		StoreType:      manifest.Keys,
-		VaultType:      vaultType,
+	if name != "" && secretStore != "" {
+		errMessage := "cannot specify vault and secret store simultaneously. Please choose one option"
+		logger.Error(errMessage)
+		return errors.InvalidParameterError(errMessage)
 	}
-	c.mux.Unlock()
-
-	logger.Info("key store created successfully")
-	return nil
-}
-
-func (c *Connector) createKeyStore(storeName string, vaultType manifest.VaultType, specs interface{}) (stores.KeyStore, error) {
-	logger := c.logger.With("vault_type", vaultType, "store_name", storeName)
-
-	switch vaultType {
-	case manifest.HashicorpKeys:
-		spec := &entities.HashicorpSpecs{}
-		if err := json.UnmarshalJSON(specs, spec); err != nil {
-			errMessage := "failed to unmarshal Hashicorp key store specs"
-			logger.WithError(err).Error(errMessage)
-			return nil, errors.InvalidParameterError(errMessage)
-		}
-
-		return newHashicorpKeyStore(spec, logger)
-	case manifest.AKVKeys:
-		spec := &entities.AkvSpecs{}
-		if err := json.UnmarshalJSON(specs, spec); err != nil {
-			errMessage := "failed to unmarshal AKV key store specs"
-			logger.WithError(err).Error(errMessage)
-			return nil, errors.InvalidParameterError(errMessage)
-		}
 
-		return newAkvKeyStore(spec, logger)
-	case manifest.AWSKeys:
-		spec := &entities.AwsSpecs{}
-		if err := json.UnmarshalJSON(specs, spec); err != nil {
-			errMessage := "failed to unmarshal AWS key store specs"
-			logger.WithError(err).Error(errMessage)
-			return nil, errors.InvalidParameterError(errMessage)
-		}
-
-		return newAwsKeyStore(spec, logger)
-	case manifest.LocalKeys:
-		localKeySpecs := &entities.LocalKeySpecs{}
-		if err := json.UnmarshalJSON(specs, localKeySpecs); err != nil {
-			errMessage := "failed to unmarshal local key store specs"
-			logger.WithError(err).Error(errMessage)
-			return nil, errors.InvalidParameterError(errMessage)
+	// If vault is specified, it is a remote key store, otherwise it's a local key store
+	var store stores.KeyStore
+	var err error
+	switch {
+	case vault != "":
+		store, err = c.createKeyStore(ctx, vault)
+		if err != nil {
+			return err
 		}
+	case secretStore != "":
+		// TODO: Uncomment when authManager no longer a runnable
+		// permissions := c.authManager.UserPermissions(userInfo)
```

Comment:
What would it be the propose of this action here?

Reply:
Add RBAC for vaults (Resource)